### DISCUSSION TITLE
Error: cannot convert from 'System.IO.StreamWriter' to 'System.IO.Stream'

### DISCUSSION THREAD

**Discussion by daryl-williams** [SixLabors/ImageSharp#2513] [Q&A] [answered]
I am trying to use the image.Save(stream, encoder) method to write an image to stdout as follows:
```
var stdout = new StreamWriter(Console.OpenStandardOutput());
stdout.AutoFlush = true;
image.Save(stdout, new JpegEncoder());
```
but I get the following error:

`error CS1503: Argument 1: cannot convert from 'System.IO.StreamWriter' to 'System.IO.Stream'`

I have searched online trying to understand the difference between System.IO.StreamWriter and System.IO.Stream without much success. I tried casting but that didn't work. Any help would be much appreciated. Is it even possible to use Image.Save in this manner?

Regards,

Daryl

**Comment by tocsoft** [comment, marked as answer]
This should work (untested).
```c#
var stdout = Console.OpenStandardOutput()
image.Save(stdout, new JpegEncoder());
stdout.Flush()
```
Ths issue you are hitting is that you are using the API/streams completly wrong, You are trying to pass a `StreamWriter` (which is a type for simplifying wrting text to a stream) instead of the stream itself.

**Reply by daryl-williams** [reply]
Worked like a charm! Thanks, much appreciated.

D
